src/core/src/main/java/org/locationtech/geogig/data/EPSGBoundsCalc.java

```diff
@@ -0,0 +1,130 @@
+        }
+
+        Collection<? extends GeographicExtent> geographicElements;
+        geographicElements = domainOfValidity.getGeographicElements();
+
+        GeographicExtent geographicExtent = geographicElements.iterator().next();
+        GeographicBoundingBox geographicBoundingBox = (GeographicBoundingBox) geographicExtent;
+
+        double minx = geographicBoundingBox.getWestBoundLongitude();
+        double miny = geographicBoundingBox.getSouthBoundLatitude();
+        double maxx = geographicBoundingBox.getEastBoundLongitude();
+        double maxy = geographicBoundingBox.getNorthBoundLatitude();
+
+        CoordinateReferenceSystem wgs84LongFirst;
+
+        wgs84LongFirst = CRS.decode("EPSG:4326", true);
+        CoordinateOperationFactory coordOpFactory = CRS.getCoordinateOperationFactory(true);
+        CoordinateOperation op = coordOpFactory.createOperation(wgs84LongFirst,crs);
+
+        ReferencedEnvelope refEnvelope = new ReferencedEnvelope(minx, maxx, miny, maxy, wgs84LongFirst);
+        GeneralEnvelope genEnvelope = CRS.transform(op, refEnvelope);
+
+        double xmax = genEnvelope.getMaximum(0);
+        double ymax = genEnvelope.getMaximum(1);
+        double xmin = genEnvelope.getMinimum(0);
+        double ymin = genEnvelope.getMinimum(1);
+
+        Envelope envelope = new Envelope(xmin, xmax, ymin, ymax);
+        return envelope;
+    }
+
+    /**
+     * Search for the given CRS (EPSG code), return the bounds (domain of validity)
+     * @param refId the input CRS
+     * @return projectionBounds an Envelope describing the CRS bounds, throws
+     * NoSuchAuthorityException if the
+     * CRS is not found
+     */
+    public Envelope findCode(String refId) throws Exception {
+        Envelope projectionBounds = null;
+        CoordinateReferenceSystem crs = null;
+
+        CRSAuthorityFactory authorityFactory = CRS.getAuthorityFactory(true);
+
+        crs = authorityFactory.createCoordinateReferenceSystem(refId);
+
+        projectionBounds = getExtents(crs);
+        return projectionBounds;
+    }
+
+    /**
+     * Search for the given CRS (EPSG code), return the bounds (domain of validity)
+     * @param featureType the RevFeatureType to find the CRS bounds of
+     * @return bounds an Envelope describing the CRS bounds, throws NoSuchAuthorityException if the
+     * CRS is not found
+     */
+    public Envelope getCRSBounds(Optional<RevFeatureType> featureType) throws Exception {
+        Envelope bounds;
+        Set<ReferenceIdentifier> code = null;
+
+        List<PropertyDescriptor> descList = featureType.get().descriptors().asList();
+        for (PropertyDescriptor desc : descList) {
+            if (desc instanceof GeometryDescriptor) {
+                code = ((GeometryDescriptor) desc).getCoordinateReferenceSystem().getIdentifiers();
+            }
+        }
+        bounds = findCode(code.toString().replaceAll("[\\[\\]]",""));
```

Comment:
Agreed. Not sure the best approach here, but it seems like you want to build a String that looks like `EPSG:4326` from the GeometryDescriptor CRS, so you can pass it to the `findCode()` method to get the Envelope. I haven't looked, but maybe `getCodeSpace()` and `getCode()` on the ReferenceIdentifier can be used to build the desired String sent into `findCode()`?